components/dashboard/src/hooks/useStripe.ts

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { useEffect, useState } from "react";
+import { getGitpodService } from "../service/service";
+import { useLocation } from "react-router";
+import { AttributionId } from "@gitpod/gitpod-protocol/lib/attribution";
+
+interface hasId {
+    id: string;
+}
+
+type PendingStripeSubscription = { pendingSince: number };
+
+type returnType = [
+    boolean,
+    string | undefined,
+    boolean,
+    boolean,
+    string | undefined,
+    number | undefined,
+    (newLimit: number) => Promise<void>,
+];
```

Comment:
Oof, this is a bit unreadable and looks a bit error prone 🙈 I'll try to track down what all these things are (and if they're all really needed), but if we do indeed need all of these, what do you think of returning a custom shape (with clearly named attributes) instead of an array with implicit positions?

Reply:
Hook removed.